mozilla-release/browser/components/shell/nsWindowsShellService.cpp

```diff
@@ -367,6 +367,40 @@ IsDefaultBrowserWin8(bool aCheckAllTypes, bool* aIsDefaultBrowser)
 }
 
 /*
+ * CLIQZ. Fix acidentally replacing of FirefoxHTML and FirefoxURL
+ * keys in registry for HKCU. Firefox will show "set as default"
+ * dialog to fix this problem on their side. Same for CLIQZ
+ * everything will be ok after
+*/
+static void FixCliqzRegistry()
+{
+  HKEY theKey;
+
+  DWORD res = ::RegOpenKeyW(HKEY_CURRENT_USER,
+                            L"Software\\Classes\\FirefoxHTML",
+                            &theKey);
+  if (REG_SUCCEEDED(res)) {
+    wchar_t currValue[MAX_BUF];
+    ::ZeroMemory(currValue, sizeof(currValue));
+    DWORD len = sizeof currValue;
+    res = ::RegQueryValueExW(theKey, L"", nullptr, nullptr,
+                             (LPBYTE)currValue, &len);
+    // If text in Firefox key started with CLIQZ - need to remove key
+    if (!wcsncmp(L"CLIQZ", currValue, 5)) {
+      ::SHDeleteKey(theKey, "");
+      ::RegCloseKey(theKey);
+      // one more key must be deleted
+      ::RegOpenKeyW(HKEY_CURRENT_USER,
+                    L"Software\\Classes\\FirefoxURL",
+                    &theKey);
+      ::SHDeleteKey(theKey, "");
+    }
+
+    ::RegCloseKey(theKey);
```

Comment:
Yep, but why FF coders doesn't use it here? I think here could be some limitation in use xpcom here